src/Approximations/template_directions.jl

```diff
@@ -268,45 +500,45 @@ Then the Cartesian components of each direction are obtained with
 The integer passed as an argument is used to discretize ``φ``:
 
 ```jldoctest; filter = r"2246[0-9]*e-16"
-julia> using LazySets.Approximations: PolarDirections
-
 julia> pd = PolarDirections(2)
-PolarDirections{Float64}(2, Array{Float64,1}[[1.0, 0.0], [-1.0, 1.22465e-16]])
+PolarDirections{Float64,Array{Float64,1}}(2, Array{Float64,1}[[1.0, 0.0], [-1.0, 1.2246467991473532e-16]])
 
-julia> pd.Nφ
+julia> length(pd)
 2
 ```
 """
-struct PolarDirections{N<:AbstractFloat} <: AbstractDirections{N}
+struct PolarDirections{N<:AbstractFloat, VN<:AbstractVector{N}} <: AbstractDirections{N, VN}
     Nφ::Int
-    stack::Vector{Vector{N}} # stores the polar directions
+    stack::Vector{VN} # stores the polar directions
+end
 
-    function PolarDirections{N}(Nφ::Int) where {N<:AbstractFloat}
-        if Nφ <= 0
-            throw(ArgumentError("Nφ = $Nφ is invalid; it shoud be at least 1"))
-        end
-        stack = Vector{Vector{N}}()
-        # discretization of the polar angle
-        φ = range(N(0.0), N(2*pi), length=Nφ+1)
+# convenience constructors
+PolarDirections(Nφ::Int) = PolarDirections{Float64, Vector{Float64}}(Nφ)
+PolarDirections{N}(Nφ::Int) where {N} = PolarDirections{N, Vector{N}}(Nφ)
 
-        for φᵢ in φ[1:Nφ]  # skip last (repeated) angle
-            d = N[cos(φᵢ), sin(φᵢ)]
-            push!(stack, d)
-        end
-        return new{N}(Nφ, stack)
+function PolarDirections{N, Vector{N}}(Nφ::Int) where {N}
+    if Nφ <= 0
+        throw(ArgumentError("Nφ = $Nφ is invalid; it shoud be at least 1"))
+    end
+    stack = Vector{Vector{N}}(undef, Nφ)
+    φ = range(N(0), N(2*pi), length=Nφ+1)  # discretization of the polar angle
+
+    @inbounds for i in 1:Nφ  # skip last (repeated) angle
+        stack[i] = N[cos(φ[i]), sin(φ[i])]
     end
+    return PolarDirections{N, Vector{N}}(Nφ, stack)
 end
```

Comment:
Is there a reason for making this an outer constructor?

Reply:
so it would be better to use StaticArrays for this template, since their size is known from the type (same for SphericalDirections). we can add one `PolarDirections{N, SVector{...}` in the future.